vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -2013,3 +2051,129 @@ resource "vsphere_virtual_machine" "vm" {
 		os.Getenv("VSPHERE_USE_LINKED_CLONE"),
 	)
 }
+
+func testAccResourceVSphereVirtualMachineConfigSlashNetwork() string {
+	return fmt.Sprintf(`
+variable "datacenter" {
+  default = "%s"
+}
+
+variable "hosts" {
+  default = [
+    "%s",
+    "%s",
+    "%s",
+  ]
+}
+
+variable "switch_nic" {
+  default = "%s"
+}
+
+variable "cluster" {
+  default = "%s"
+}
+
+variable "resource_pool" {
+  default = "%s"
+}
+
+variable "network_label" {
+  default = "%s"
+}
+
+variable "ipv4_address" {
+  default = "%s"
+}
+
+variable "ipv4_prefix" {
+  default = "%s"
+}
+
+variable "ipv4_gateway" {
+  default = "%s"
+}
+
+variable "datastore" {
+  default = "%s"
+}
+
+variable "template" {
+  default = "%s"
+}
+
+variable "linked_clone" {
+  default = "%s"
+}
+
+data "vsphere_datacenter" "datacenter" {
+  name = "${var.datacenter}"
+}
+
+data "vsphere_host" "host" {
+  count         = "${length(var.hosts)}"
+  name          = "${var.hosts[count.index]}"
+  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
+}
+
+resource "vsphere_host_virtual_switch" "switch" {
+  count          = "${length(data.vsphere_host.host.*.id)}"
+  name           = "vSwitchTerraformTest"
+  host_system_id = "${data.vsphere_host.host.*.id[count.index]}"
+
+  network_adapters = ["${var.switch_nic}"]
+
+  active_nics  = ["${var.switch_nic}"]
+  standby_nics = []
+}
+
+resource "vsphere_host_port_group" "pg" {
+  count               = "${length(data.vsphere_host.host.*.id)}"
+  name                = "${var.network_label}"
+  host_system_id      = "${data.vsphere_host.host.*.id[count.index]}"
+  virtual_switch_name = "${vsphere_host_virtual_switch.switch.*.name[count.index]}"
+}
+
+resource "vsphere_virtual_machine" "vm" {
+  name          = "terraform-test"
+  datacenter    = "${var.datacenter}"
+  cluster       = "${var.cluster}"
+  resource_pool = "${var.resource_pool}"
+
+  vcpu   = 2
+  memory = 1024
+
+  network_interface {
+    label              = "${vsphere_host_port_group.pg.0.name}"
+    ipv4_address       = "${var.ipv4_address}"
+    ipv4_prefix_length = "${var.ipv4_prefix}"
+    ipv4_gateway       = "${var.ipv4_gateway}"
+  }
+
+  disk {
+    datastore = "${var.datastore}"
+    template  = "${var.template}"
+    iops      = 500
+  }
+
+  depends_on = ["vsphere_host_port_group.pg"]
```

Comment:
I'd have thought this should be implied by `network_interface.0.label` being set to `"${vsphere_host_port_group.pg.0.name}"`?

Reply:
We wait on all resources here because while we are only referencing the first resource for the network name, the VM could be created on any of the three hosts, hence it's important to ensure that the network is available on all three before proceeding with creation of the VM.